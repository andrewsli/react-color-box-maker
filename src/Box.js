import React, { Component } from "react";

class Box extends Component {
  constructor(props) {
    super(props);
    this.removeBox = this.removeBox.bind(this);
  }

  //call parent removeBox function, pass in current box ID
  removeBox() {
    this.props.removeThisBox(this.props.id)
  }

  render() {
    let { height, width, backgroundColor } = this.props;
    let styles = { height, width, backgroundColor, margin: "3px", display: "inline-block"}
    return (
      <div>
        <div style={styles} className="box">
        </div>
        <button onClick={this.removeBox}>X</button>
      </div>

    )
  }
}


export default Box;