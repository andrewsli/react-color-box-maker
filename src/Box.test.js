import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Box from "./Box";


it("renders using shallow", function() {
    shallow(<Box />);
});


it("matches snapshot", function() {
    let wrapper = shallow(<Box />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});