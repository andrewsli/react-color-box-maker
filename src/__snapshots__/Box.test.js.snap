// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="box"
    style={
      Object {
        "backgroundColor": undefined,
        "display": "inline-block",
        "height": undefined,
        "margin": "3px",
        "width": undefined,
      }
    }
  />
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
